examples/proportion_test.py

```diff
@@ -0,0 +1,134 @@
+
+This example answers this question by estimating a logistic regression model where the
+covariates are whether the customer got called and their gender. We place a multivariate
+normal prior on the regression coefficients. We report the 95% highest posterior
+density interval for the effect of making a call.
+"""
+
+
+import argparse
+import os
+from typing import Any, Callable, Tuple
+
+import jax.numpy as np
+import numpyro
+import numpyro.distributions as dist
+import pandas as pd
+from jax import random
+from jax.scipy.special import expit
+from numpyro.diagnostics import hpdi
+from numpyro.infer import MCMC, NUTS
+from scipy import stats
+
+
+def make_dataset() -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Make simulated dataset where potential customers who get a 
+    sales calls have ~2% higher chance of making another purchase.
+    """
+    made_purchase_got_called = stats.bernoulli.rvs(0.084, size=51342)
+    made_purchase_no_calls = stats.bernoulli.rvs(0.061, size=48658)
+    
+    made_purchase = np.concatenate([made_purchase_got_called, made_purchase_no_calls])
+    
+    is_female = stats.bernoulli.rvs(0.5, size=100000)
+    got_called = np.concatenate([np.ones(51342), np.zeros(48658)])
+    design_matrix = np.hstack([np.ones((100000, 1)),
+                               got_called.reshape(-1, 1),
+                               is_female.reshape(-1, 1)])
+    
+    return design_matrix, made_purchase
+
+
+def model(design_matrix: np.ndarray, outcome: np.ndarray = None) -> None:
+    """
+    Model definition: Log odds of making a purchase is a linear combination
+    of covariates. Specify aNormal prior over regression coefficients.
+    :param design_matrix: Covariates. All categorical variables have been one-hot
+        encoded.
+    :param outcome: Binary response variable. In this case, whether or not the
+        customer made a purchase.
+    """
+    
+    # Use multivariate normal prior in case we want to add more covariates later.
+    beta = numpyro.sample('coefficients', dist.MultivariateNormal(loc=0.,
+                                                                  covariance_matrix=np.eye(design_matrix.shape[1])))
+    logits = design_matrix.dot(beta)
+    
+    with numpyro.plate('data', design_matrix.shape[0]):
+            numpyro.sample('obs', dist.Bernoulli(logits=logits), obs=outcome)
+    
+
+def print_results(coef: np.ndarray, interval_size: float = 0.95) -> None:
+    """
+    Print the confidence interval for the effect size with interval_size
+    probability mass.
+    """
+
+    baseline = hpdi(coef[:, 0], prob=0.95)
+    effect_size = hpdi(coef[:, 1], prob=0.95)
+    
+    baseline_response = expit(baseline)
+    response_with_calls = expit(baseline + effect_size)
+    
+    impact_on_probability = response_with_calls - baseline_response
+    
+    print(f"There is a {interval_size * 100}% probability that calling customers "
+          "increases the chance they'll make a purchase by "
```

Comment:
This looks good to me since we constructed the dataset, but users extrapolating it to more complex examples with multiple predictors might be problematic. @fehiepsi, @martinjankowiak: WDYT?

Reply:
i think this is fine for this self-contained example 👍